malta.ld

```diff
@@ -26,7 +26,7 @@ SECTIONS
     . = ALIGN(4096);
   } : text
 
-  .data ADDR(.text) + SIZEOF (.text):
+  .data :
```

Comment:
The reason for this change is that if we "manually" compute section locations, the linker heuristic responsible for placing orphaned sections decides to place them after `.text`, so such section and our `.data` overlap. The link sets need to be an orphaned section, because only then will the linker provide `__start_*` and `__end_*` symbols.

For consistence, I also removed this address computation from `.bss`, I don't think we need to explicitly specify the address - the linker will use the correct one anyway.